cache/contenthash/checksum.go

```diff
@@ -648,6 +684,82 @@ func shouldIncludePath(
 	return true, nil
 }
 
+func wildcardPrefix(root *iradix.Node, p string) (string, []byte, bool, error) {
+	// For consistency with what the copy implementation in fsutil
+	// does: split pattern into non-wildcard prefix and rest of
+	// pattern, then follow symlinks when resolving the non-wildcard
+	// prefix.
+
+	d1, d2 := splitWildcards(p)
+	if d1 == "/" {
+		return "", nil, false, nil
+	}
+
+	k, cr, err := getFollowLinks(root, convertPathToKey([]byte(d1)), true)
```

Comment:
this could use `getFollowLinksWalk` so you can reuse `linkWalked` without double counting.

Reply:
Updated to use `getFollowLinksWalk`.